sentry-spring-boot-starter/src/main/java/io/sentry/spring/boot/ContextTagsEventProcessor.java

```diff
@@ -0,0 +1,41 @@
+package io.sentry.spring.boot;
+
+import io.sentry.EventProcessor;
+import io.sentry.SentryEvent;
+import io.sentry.SentryOptions;
+import io.sentry.util.CollectionUtils;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.MDC;
+
+/**
+ * Attaches context tags defined in {@link SentryOptions#getContextTags()} from {@link MDC} to
+ * {@link SentryEvent#getTags()}.
+ */
+public final class ContextTagsEventProcessor implements EventProcessor {
+  private final SentryOptions options;
+
+  public ContextTagsEventProcessor(final @NotNull SentryOptions options) {
+    this.options = options;
+  }
+
+  @Override
+  public @NotNull SentryEvent process(
+      @NotNull SentryEvent event, @Nullable Map<String, Object> hint) {
+    final Map<String, String> contextMap = MDC.getCopyOfContextMap();
```

Comment:
Perhaps we could filter only to events with mechanism type `HandlerExceptionResolver`, otherwise for events send from logback this logic will be applied twice.

On the other hand, if we don't filter by mechanism type, context tags are also applied to events triggered manually with `Sentry.captureException()`.

Reply:
Is it bad if direct captures also get the MDC context added to the tags?

Reply:
You can also filter by `event.getLogger` since it sets `event.setLogger(record.getLoggerName());` for JUL and Logback, but IMO, we should add MDC anyways, if they don't want, they won't add in the `options.getContextTags()` in the first place.

Reply:
So if i am getting it right, we are keeping it as it is, correct?

Reply:
Yes, but that's just my opinion, I know that we should always be skeptical about adding a lot of tags since high cardinality data is more expensive, so please double-check with @bruno-garcia

Reply:
I'd keep it this way to add MDC tags regardless of logging framework.

The way it is implemented it'll just override existing tags and should just work or de we expect any problems regarding MDC being thread local?

I think we also need PRs #2052 , #2054 and #2057 in case we're not dealing with spring.

Reply:
If someone does not use Spring Boot Starter, then they have to configure Logback explicitly in `logback.xml` - meaning they have to provide configuration - so it's somewhat expected that context tags set in `logback.xml` are taken into account. Then I believe we do not need any other change.